src/app/clusters/commissioner-control-server/commissioner-control-server.h

```diff
@@ -0,0 +1,99 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace CommissionerControl {
+
+struct CommissioningApprovalRequest
+{
+    uint64_t requestId;
+    VendorId vendorId;
+    uint16_t productId;
+    NodeId clientNodeId;
+    FabricIndex fabricIndex;
+    Optional<chip::CharSpan> label;
+};
+
+struct CommissioningWindowParams
+{
+    uint32_t iterations;
+    uint16_t commissioningTimeout;
+    uint16_t discriminator;
+    ByteSpan PAKEPasscodeVerifier;
+    ByteSpan salt;
+};
+
+struct CommissionNodeInfo
+{
+    CommissioningWindowParams params;
+    Optional<ByteSpan> ipAddress;
+    Optional<uint16_t> port;
```

Comment:
When do we ever commission specifically by IP address? How is this going to be used?

Reply:
If the client provides the ipAddress and port, the application can optionally choose to use 'pairing already-discovered' instead of 'pairing code' to reverse pair the device. But this is up to the application, this just provide the flexibility.